### ISSUE TITLE
GUI improvements

### ISSUE THREAD

**Issue by p3lim** [PrinceOfAmber/Storage-Network#409]
This mod is very nice feature-wise, but the GUI needs a lot of love.
Some suggestions:

- Grids should look like chests
  - which includes the font rendering
- Grids should have scrollbars
- Grids should have a toggle to auto-focus the search bar
- Grids should have a scaling option (e.g. 3/6/9/12 rows)
- Everything should be centered/aligned
  - buttons and such
- Every GUI should have a title (e.g. name of the block in the top-left)
- Priority inputs should be input boxes, not a counter with +/- buttons
  - sidenote: given the name "simple storage network", having inverted priorities is confusing, higher number should be higher priority
- Storage Request Table should have a tooltip on the crafting output
  - there is no indicator what left/right clicking does without trying, wasting resources

For just general design I'd look at how AE/RS/Tom's does it, they've pretty much just re-used the layout/design of the vanilla chest and added buttons/scrollbars to the sides, giving it a uniform look.

Aside from the font rendering, this should be reasonably simple job to implement, as it's just some reordering/offset adjustments and some minor texture work.

**Comment by Lothrazar** [comment]
I agree, the GUI design is ancient, and i havent changed it once since taking over the mod from the other author 5+ years ago. and thanks for writing it down with all the detail.  Once i get some bugs cleaned up ill keep this near the top of my list

**Comment by Lothrazar** [comment]
still planning to look at this . 
Also related: gui improvements https://github.com/Lothrazar/Storage-Network/issues/357

**Comment by Lothrazar** [comment]
Priority ref https://github.com/Lothrazar/Storage-Network/issues/396

**Comment by Lothrazar** [comment]
if the toggling of search autofocus is added to GUI make sure to delete the config for it "   ENABLEAUTOSEARCHFOCUS = COMMON_BUILDER.comment("\r\n Set to false to disable the automatic focus of the searchbar - client gui screen")
        .define("enableAutoSearchFocus", true);"

**Comment by GhostGH** [comment]
so, any update on this?

Could use Applied Energistics 2 (open-source) as an example. Their ME Terminal GUI:

![Image](https://github.com/user-attachments/assets/8f931dd8-04a9-4f59-bb96-0dcb9994f4af)
